controllers/remove_process_groups.go

```diff
@@ -277,6 +268,30 @@ func includeProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler,
 	return nil
 }
 
+func getIncludedProcesses(cluster *fdbv1beta2.FoundationDBCluster, removedProcessGroups map[string]bool, fdbProcessesToInclude *[]fdbv1beta2.ProcessAddress, processGroups *[]*fdbv1beta2.ProcessGroupStatus) (bool, error) {
+	fdbVersion, err := fdbv1beta2.ParseFdbVersion(cluster.Spec.Version)
+	if err != nil {
+		return false, err
+	}
+
+	hasStatusUpdate := false
+	for _, processGroup := range cluster.Status.ProcessGroups {
+		if processGroup.IsMarkedForRemoval() && removedProcessGroups[processGroup.ProcessGroupID] {
+			if fdbVersion.IsAtLeast(fdbv1beta2.Versions.LatestFdbVersion) {
+				*fdbProcessesToInclude = append(*fdbProcessesToInclude, fdbv1beta2.ProcessAddress{StringAddress: processGroup.GetExclusionString()})
```

Comment:
We should also add in this case the IP addresses of the process group otherwise we might run in the following case:

1. Operator excludes process group
2. Upgrade to 7.0.0+ is started (and locality based exclusion is enabled)
3. Process group get's excluded again with the locality string
4. Process group gets removed and we only include the locality string and not the IPs (leaving them excluded in the cluster).